tests/pass/permissive_provenance_sb.rs

```diff
@@ -0,0 +1,62 @@
+// compile-flags: -Zmiri-permissive-provenance
+
+
+// random agglomeration test
+use core::cell::UnsafeCell;
+
+fn not_so_innocent(x: &mut UnsafeCell<u32>) {
+    let x_raw: *mut UnsafeCell<u32> = x as *mut UnsafeCell<u32>;
+    x_raw as usize;
+}
+
+fn main() {
+    unsafe { 
+        let mut c = UnsafeCell::new(42u32);
+
+        let x_unique1: &mut UnsafeCell<u32> = &mut c;
+        // [..., Unique(1)]
+        
+        let x_raw1: *mut UnsafeCell<u32> = x_unique1 as *mut UnsafeCell<u32>;
+        let x_raw1_usize = x_raw1 as usize;
+        // [..., Unique(1), SharedRW(2*)]
+        // * denotes exposed
+        
+        let x_unique2: &mut UnsafeCell<u32> = &mut *x_raw1;
+        // [.., Unique(1), SharedRW(2*), Unique(3)]
+        
+        not_so_innocent(x_unique2);
```

Comment:
Can you make this `if innocent { not_so_innocent }`, add a `innocent: bool` parameter to this function, and then run this entire test twice, for both values of `innocent`?